File: app/models/grade.rb
class Grade < ActiveRecord::Base
  belongs_to :user
  belongs_to :course
  belongs_to :weight

  def percentage
    weight = self.weight
    weighted_sum = weight.grade_score_total
    percentage = (self.score / weighted_sum)*(weight.percentage)

    percentage.round(2)
  end

end

File: frontend/app/models/grade.js
import DS from 'ember-data';

export default DS.Model.extend({
  user: DS.belongsTo('user'),
  course: DS.belongsTo('course', {async: true}),
  weight: DS.belongsTo('weight', {async: true}),
  name: DS.attr('string'),
  score: DS.attr('number'),
  scoreTotal: DS.attr('number'),
  createdAt: DS.attr('date'),

  // percentage: function() {
  //   var weightedSum = this.get('weight.scoreTotalSum');
  //   var percentage = (this.get('score')/weightedSum)*(this.get('weight.percentage'))
  //   return percentage.toFixed(2);
  // }.property('score', 'weight.scoreTotalSum', 'weight.percentage')

});
